UniswapV3Adapter/contracts/UniswapV3Adapter.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./UniswapV3likeAdapter.sol";
+
+interface IUniV3Factory {
+    function feeAmountTickSpacing(uint24) external view returns (int24);
+
+    function getPool(
+        address,
+        address,
+        uint24
+    ) external view returns (address);
+}
+
+contract UniswapV3Adapter is UniswapV3likeAdapter {
+    using SafeERC20 for IERC20;
+
+    address immutable FACTORY;
+    mapping(uint24 => bool) public isFeeAmountEnabled;
+    uint24[] public feeAmounts;
+
+    constructor(
+        string memory _name,
+        uint256 _swapGasEstimate,
+        uint256 _quoterGasLimit,
+        address _quoter,
+        address _factory
+    ) UniswapV3likeAdapter(_name, _swapGasEstimate, _quoter, _quoterGasLimit) {
```

Comment:
Severity: Low
Confidence: Medium

```
UniswapV3likeAdapter.setQuoter(address).newQuoter lacks a zero-check on :
                - quoter = newQuoter
                
UniswapV3Adapter.constructor(string,uint256,uint256,address,address)._factory lacks a zero-check on :
                - FACTORY = _factory
```
                
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#missing-zero-address-validation

Recommendation: Use zero address check for `_quoter` in `UniswapV3LikeAdapter.sol` constructor and zero address check for `_factory` in `UniswapV3Adapter.sol` constructor.